src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowRun.java

```diff
@@ -371,6 +391,22 @@ public void doKill() {
         return env;
     }
 
+    public void setAllowTerm(boolean b) {
+        this.allowTerm = b;
+    }
+
+    public boolean hasAllowTerm() {
+        return allowTerm && !allowKill;
```

Comment:
No, you should be allowed to retry `term` even if `kill` has been offered. (As you can already do via hyperlinks.)